### ISSUE TITLE
Avoid scaling textures unless needed

### ISSUE THREAD

**Issue by lhofhansl** [minetest/minetest#6580]
Currently we always scale all textures to at least a resolution of 64.
That is done to reduce blur with bi/trilinear filtered is on, or autoscaling is enabled (default off).

When both filtering and autoscaling are off, this simply wastes client memory for no benefit at all.

This simple fix avoids the scaling unless needed.
(the autoscaling part is only used in nodedef.cpp and not changed)

```
diff --git a/src/client/tile.cpp b/src/client/tile.cpp
index 9321a458..d0b9d81b 100644
--- a/src/client/tile.cpp
+++ b/src/client/tile.cpp
@@ -1777,7 +1777,8 @@ bool TextureSource::generateImagePart(std::string part_of_name,
                         * mix high- and low-res textures, or for mods with least-common-denominator
                         * textures that don't have the resources to offer high-res alternatives.
                         */
-                       s32 scaleto = g_settings->getS32("texture_min_size");
+                       const bool filter = m_setting_trilinear_filter || m_setting_bilinear_filter;
+                       const s32 scaleto = filter ? g_settings->getS32("texture_min_size") : 1;
                        if (scaleto > 1) {
                                const core::dimension2d<u32> dim = baseimg->getDimension();
```

**Comment by paramat** [comment]
> Currently we always scale all textures to at least a resolution of 64.

Wow.
Good find, well worth fixing, most textures are 16px.

**Comment by lhofhansl** [comment]
Small enough to just make the change, or does it need a PR?

**Comment by paramat** [comment]
PR is best.

**Comment by lhofhansl** [comment]
Fixed by #6586
